src/test/scala/chiselTests/PrintableSpec.scala

```diff
@@ -261,4 +261,208 @@ class PrintableSpec extends AnyFlatSpec with Matchers {
       """printf(clock, UInt<1>("h1"), "adieu AnonymousBundle(foo -> %d, bar -> %d)", myBun.foo, myBun.bar) : farewell"""
     )
   }
+
+  // Unit tests for cf
+  it should "handle Printable and String concatination with cf format specifier" in {
+    class MyModule extends BasicTester {
+      printf(cf"First " + PString("Second ") + "Third")
+    }
+    val firrtl = ChiselStage.emitChirrtl(new MyModule)
+    getPrintfs(firrtl) match {
+      case Seq(Printf("First Second Third", Seq())) =>
+      case e                                        => fail()
+    }
+  }
+  it should "call toString on non-Printable objects with cf format specifier" in {
+    class MyModule extends BasicTester {
+      val myInt = 1234
+      printf(cf"myInt = $myInt")
+    }
+    val firrtl = ChiselStage.emitChirrtl(new MyModule)
+    getPrintfs(firrtl) match {
+      case Seq(Printf("myInt = 1234", Seq())) =>
+      case e                                  => fail()
+    }
+  }
+  it should "generate proper printf for simple Decimal printing with cf format specifier" in {
+    class MyModule extends BasicTester {
+      val myWire = WireDefault(1234.U)
+      printf(cf"myWire = ${Decimal(myWire)}")
+    }
+    val firrtl = ChiselStage.emitChirrtl(new MyModule)
+    getPrintfs(firrtl) match {
+      case Seq(Printf("myWire = %d", Seq("myWire"))) =>
+      case e                                         => fail()
+    }
+  }
+  it should "correctly escape percent with cf format specifier" in {
+    class MyModule extends BasicTester {
+      printf(cf"%")
+    }
+    val firrtl = ChiselStage.emitChirrtl(new MyModule)
+    getPrintfs(firrtl) match {
+      case Seq(Printf("%%", Seq())) =>
+      case e                        => fail()
+    }
+  }
+  it should "correctly emit tab with cf format specifier" in {
+    class MyModule extends BasicTester {
+      printf(cf"\t")
+    }
+    val firrtl = ChiselStage.emitChirrtl(new MyModule)
+    getPrintfs(firrtl) match {
+      case Seq(Printf("\\t", Seq())) =>
+      case e                         => fail()
+    }
+  }
+  it should "support names of circuit elements including submodule IO with cf format specifier" in {
+    // Submodule IO is a subtle issue because the Chisel element has a different
+    // parent module
+    class MySubModule extends Module {
+      val io = IO(new Bundle {
+        val fizz = UInt(32.W)
+      })
+    }
+    class MyBundle extends Bundle {
+      val foo = UInt(32.W)
+    }
+    class MyModule extends BasicTester {
+      override def desiredName: String = "MyModule"
+      val myWire = Wire(new MyBundle)
+      val myInst = Module(new MySubModule)
+      printf(cf"${Name(myWire.foo)}")
+      printf(cf"${FullName(myWire.foo)}")
+      printf(cf"${FullName(myInst.io.fizz)}")
```

Comment:
```suggestion
      printf(cf"${myWire.foo}%n")
      printf(cf"${myWire.foo}%N")
      printf(cf"${myInst.io.fizz}%N")
```

Reply:
As discussed - have a version of this in one the tests below - this particular test is the "p" version - just replacing with cf. 

That being said - still more tests will be added to cover additional cases.

Reply:
This scenario is already added in the spec file - marking resolved

Reply:
https://github.com/chipsalliance/chisel3/blob/c3b1412e0932c620dcb6cce38bbe6988673fc355/src/test/scala/chiselTests/PrintableSpec.scala#L284